impl/maven-impl/src/test/java/org/apache/maven/impl/DefaultPluginXmlFactoryTest.java

```diff
@@ -0,0 +1,297 @@
+                                .build())
+                        .build()))
+                .actual();
+        assertThat(exception.getMessage()).contains(unableToWritePlugin);
+        assertThat(exception.getCause()).isInstanceOf(XmlWriterException.class);
+        assertThat(exception.getCause().getCause().getMessage()).isEqualTo(ioEx);
+        assertThat(exception.getCause().getMessage()).isEqualTo(unableToWritePlugin);
+    }
+
+    @Test
+    void writeWithNoTargetThrowsIllegalArgumentException() {
+        assertThat(assertThrows(
+                                IllegalArgumentException.class,
+                                () -> defaultPluginXmlFactory.write(XmlWriterRequest.<PluginDescriptor>builder()
+                                        .content(PluginDescriptor.newBuilder()
+                                                .name("No Output Plugin")
+                                                .build())
+                                        .build()))
+                        .getMessage())
+                .isEqualTo("writer, outputStream or path must be non null");
+    }
+
+    @Test
+    void readMalformedXmlThrowsXmlReaderException() {
+        XmlReaderException exception = assertThatExceptionOfType(XmlReaderException.class)
+                .isThrownBy(() -> defaultPluginXmlFactory.read(XmlReaderRequest.builder()
+                        .inputStream(new ByteArrayInputStream("<plugin><name>Broken Plugin".getBytes()))
+                        .build()))
+                .actual();
+        assertThat(exception.getMessage()).contains("Unable to read plugin");
+        assertThat(exception.getCause()).isInstanceOf(WstxEOFException.class);
+    }
+
+    @Test
+    void locateExistingPomWithFilePathShouldReturnSameFileIfRegularFile() throws IOException {
+        Path pomFile = Files.createTempFile(tempDir, "pom", ".xml");
+        DefaultModelProcessor processor = new DefaultModelProcessor(mock(ModelXmlFactory.class), List.of());
+        assertThat(processor.locateExistingPom(pomFile)).isEqualTo(pomFile);
+    }
+
+    @Test
+    void readFromUrlParsesPluginDescriptorCorrectly() throws Exception {
+        Path xmlFile = tempDir.resolve("plugin.xml");
+        Files.write(xmlFile, SAMPLE_PLUGIN_XML.getBytes());
+        PluginDescriptor descriptor = defaultPluginXmlFactory.read(XmlReaderRequest.builder()
+                .inputStream(xmlFile.toUri().toURL().openStream())
+                .build());
+        assertThat(descriptor.getName()).isEqualTo(NAME);
+        assertThat(descriptor.getGroupId()).isEqualTo("org.example");
+        assertThat(descriptor.getArtifactId()).isEqualTo("sample-plugin");
+        assertThat(descriptor.getVersion()).isEqualTo("1.0.0");
+    }
+
+    @Test
+    void testReadWithPath() throws Exception {
+        Path tempPath = Files.createTempFile("plugin", ".xml");
+        Files.writeString(tempPath, "<plugin/>");
+        XmlReaderRequest request = spy(XmlReaderRequest.class);
```

Comment:
<img width="1900" alt="image" src="https://github.com/user-attachments/assets/863fad68-0949-40bd-bb8e-0e6a6fb73f53" />
require `spy` now, which indicates strong test, as a lot is going on.